### PR TITLE
Flags removal

### PR DESCRIPTION
**Pull request by feedforward**
#281 - Removed the flags from utils_tf.py, uses file_path.

#397 code update to remove tensorflow depreciation warnings for tutorials.

### CODE REVIEW

File: cleverhans/utils_tf.py [hunk 1]
```diff
@@ -411,3 +415,29 @@ def clip_eta(eta, ord, eps):
         factor = tf.minimum(1., eps / norm)
         eta = eta * factor
     return eta
+
+
+def tf_reduce_sum(input_tensor, axis=None, keepdims=None,
```
Comment (npapernot) [inline cleverhans/utils_tf.py]: Given that the code that this function is looking to support is only included in one tutorial folder script, I would rather modify the tutorial's logic directly at line 79 of the `tutorial_models.py` file you modified below and remove this function from the `utils_tf.py` file (we don't want to artificially extend the lifetime of features deprecated in TF by adding wrappers unless the code affected is directly part of the CleverHans core module).

Comment (feedforward) [reply cleverhans/utils_tf.py]: yeah makes sense to me.
few issues here, need your suggestion.

i was working on #397, attacks.py has the same depreciation warnings.
tf 1.4 - expects keep_dims and throws error for keepdims.
tf 1.8 - expects keepdims and throws warning for keep_dims.

are we planning to have long time support for tf 1.4 ?

Comment (npapernot) [reply cleverhans/utils_tf.py]: Thanks for pointing this out. In that case, I would recommend following the style adopted in `model_train` https://github.com/tensorflow/cleverhans/blob/master/cleverhans/utils_tf.py#L121

As far as I can tell, there are 6 lines in `attacks.py` that are potentially affected by this change in TF naming convention for keep_dims/keepdims. I would recommend modifying these 6 lines directly into something like:

```
if tf above 1.8 
    use keepdims 
else 
    use keep_dims 
    warn user that CleverHans may not support this version of TF in the future
```

Comment (feedforward) [reply cleverhans/utils_tf.py]: sure, i will write a if else condition inplace for ( utils_tf - 1, attacks.py - 6, attacks_tf.py - 6) these locations, we can remove them if and when tf 1.4 is not supported.

File: cleverhans/utils_tf.py [hunk 2]
```diff
@@ -34,7 +34,13 @@ def model_loss(y, model, mean=True):
     else:
         logits = model
 
-    out = tf.nn.softmax_cross_entropy_with_logits(logits=logits, labels=y)
+    try:
+        y = tf.stop_gradient(y)
+        out = tf.nn.softmax_cross_entropy_with_logits_v2(
+                                                logits=logits, labels=y)
+    except AttributeError:
+        out = tf.nn.softmax_cross_entropy_with_logits(
```
Comment (npapernot) [inline cleverhans/utils_tf.py]: It would be great to add a warning here saying that this version of TF will not be supported by CleverHans in the future

Comment (feedforward) [reply cleverhans/utils_tf.py]: Sounds good, will do that.

Comment (feedforward) [reply cleverhans/utils_tf.py]: Please merge the PR, after i add the warning will submit a new PR for TF depreciation warnings.(there are warnings other than keep_dims, will resolve all in new PR )

Comment (npapernot) [reply cleverhans/utils_tf.py]: Thanks, I'll do that! Looks like your last commit broke the diff though (probably due to a conflict?)

Comment (feedforward) [reply cleverhans/utils_tf.py]: Yes there was some conflict, everything is sorted now.

**Pull request merged**